common/djangoapps/entitlements/admin.py

```diff
@@ -17,6 +21,44 @@ class CourseEntitlementAdmin(admin.ModelAdmin):
     raw_id_fields = ('enrollment_course_run', 'user',)
 
 
+class CourseEntitlementSupportDetailForm(forms.ModelForm):
+    def __init__(self, *args, **kwargs):
+        super(CourseEntitlementSupportDetailForm, self).__init__(*args, **kwargs)
+
+        if self.data.get('unenrolled_run'):
+            try:
+                self.data['unenrolled_run'] = CourseKey.from_string(self.data['unenrolled_run'])
+            except InvalidKeyError:
+                raise forms.ValidationError("Cannot make a valid CourseKey from id {}!".format(self.data['unenrolled_run']))
+
+    def clean_course_id(self):
+        course_id = self.cleaned_data['unenrolled_run']
+        try:
+            course_key = CourseKey.from_string(course_id)
+        except InvalidKeyError:
+            raise forms.ValidationError("Cannot make a valid CourseKey from id {}!".format(course_id))
+
+        if not modulestore().has_course(course_key):
+            raise forms.ValidationError("Cannot find course with id {} in the modulestore".format(course_id))
+
+        return course_key
+
+    class Meta:
+        fields = '__all__'
+        model = CourseEntitlementSupportDetail
+
+
+@admin.register(CourseEntitlementSupportDetail)
+class CourseEntitlementSupportDetailAdmin(admin.ModelAdmin):
+    list_display = ('entitlement',
```

Comment:
You probably also want to add `raw_id_fields = ('support_user',)`

Reply:
Otherwise in Stage and Production the Admin page will attempt to load the whole list of users.